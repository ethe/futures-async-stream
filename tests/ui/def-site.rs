// SPDX-License-Identifier: Apache-2.0 OR MIT

#![feature(coroutines, proc_macro_hygiene, stmt_expr_attributes)]

use futures_async_stream::{for_await, stream};

#[stream(item = ())]
async fn def_site_arg1(a: Vec<u8>) {
    __arg0; //~ ERROR E0425
}

#[stream(item = ())]
async fn def_site_arg2(ref a: Vec<u8>) {
    __arg0; //~ ERROR E0425
}

#[stream(item = ())]
async fn def_site_task_context1() {
    __task_context; //~ ERROR E0425
}

#[stream(item = ())]
async fn def_site_task_context2() {
    async {}.await;
    __task_context; //~ ERROR E0425
}

#[stream(item = ())]
async fn stream() {}

#[stream(item = ())]
async fn def_site_for_await1() {
    #[for_await]
    for _ in stream() {
        &__pinned; //~ ERROR E0425
    }
}

async fn def_site_for_await2() {
    #[for_await]
    for _ in stream() {
        &__pinned; //~ ERROR E0425
    }
}

fn main() {}
