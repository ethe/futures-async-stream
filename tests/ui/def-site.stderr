error[E0425]: cannot find value `__arg0` in this scope
 --> tests/ui/def-site.rs:9:5
  |
9 |     __arg0; //~ ERROR E0425
  |     ^^^^^^ not found in this scope

error[E0425]: cannot find value `__arg0` in this scope
  --> tests/ui/def-site.rs:14:5
   |
14 |     __arg0; //~ ERROR E0425
   |     ^^^^^^ not found in this scope

error[E0425]: cannot find value `__task_context` in this scope
  --> tests/ui/def-site.rs:19:5
   |
19 |     __task_context; //~ ERROR E0425
   |     ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `__task_context` in this scope
  --> tests/ui/def-site.rs:25:5
   |
25 |     __task_context; //~ ERROR E0425
   |     ^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `__pinned` in this scope
  --> tests/ui/def-site.rs:35:10
   |
35 |         &__pinned; //~ ERROR E0425
   |          ^^^^^^^^ not found in this scope

error[E0425]: cannot find value `__pinned` in this scope
  --> tests/ui/def-site.rs:42:10
   |
42 |         &__pinned; //~ ERROR E0425
   |          ^^^^^^^^ not found in this scope
